src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -0,0 +1,173 @@
+{
+    /// <summary>
+    /// <see cref="StartupProjectRegistrar"/> is responsible for adding or removing a project from the Startup list
+    /// depending on whether the active configuration of the a project is debuggable or not.
+    /// </summary>
+    internal class StartupProjectRegistrar :
+        IDisposable
+    {
+        private readonly IVsStartupProjectsListService _startupProjectsListService;
+        private readonly IUnconfiguredProjectVsServices _projectVsServices;
+        private readonly IProjectThreadingService _threadingService;
+        private readonly IActiveConfiguredProjectSubscriptionService _activeConfiguredProjectSubscriptionService;
+        private readonly ActiveConfiguredProject<DebuggerLaunchProviders> _launchProviders;
+
+        private Guid _guid;
+        private IDisposable _evaluationSubscriptionLink;
+        private bool _isDebuggable;
+
+        internal DataFlowExtensionMethodCaller WrapperMethodCaller { get; set; }
+
+        [ImportingConstructor]
+        public StartupProjectRegistrar(
+            IUnconfiguredProjectVsServices projectVsServices,
+            SVsServiceProvider serviceProvider,
+            IProjectThreadingService threadingService,
+            IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService,
+            ActiveConfiguredProject<DebuggerLaunchProviders> launchProviders)
+        {
+            Requires.NotNull(projectVsServices, nameof(projectVsServices));
+            Requires.NotNull(serviceProvider, nameof(serviceProvider));
+            Requires.NotNull(threadingService, nameof(threadingService));
+            Requires.NotNull(activeConfiguredProjectSubscriptionService, nameof(activeConfiguredProjectSubscriptionService));
+            Requires.NotNull(launchProviders, nameof(launchProviders));
+
+            _projectVsServices = projectVsServices;
+            _startupProjectsListService = serviceProvider.GetService<IVsStartupProjectsListService, SVsStartupProjectsListService>();
+            _threadingService = threadingService;
+            _activeConfiguredProjectSubscriptionService = activeConfiguredProjectSubscriptionService;
+            _launchProviders = launchProviders;
+
+            WrapperMethodCaller = new DataFlowExtensionMethodCaller(new DataFlowExtensionMethodWrapper());
+        }
+
+        [ProjectAutoLoad]
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+        internal async Tasks.Task OnProjectFactoryCompletedAsync()
+        {
+            ConfigurationGeneral projectProperties =
+                await _projectVsServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
+            _guid = new Guid((string)await projectProperties.ProjectGuid.GetValueAsync().ConfigureAwait(false));
+            Assumes.False(_guid == Guid.Empty);
+
+            await InitializeAsync().ConfigureAwait(false);
+        }
+
+        public async Tasks.Task InitializeAsync()
```

Comment:
I think you know the reason now(I saw it in one of your PR) but for completeness sake I will answer. Task conflicts with `Microsoft.VisualStudio.Shell.Task`. Hence it needs a little more qualification to resolve ambiguity